client/hosting/server-settings/components/sftp-card/index.tsx

```diff
@@ -36,35 +35,35 @@ import { getAtomicHostingSshAccess } from 'calypso/state/selectors/get-atomic-ho
 import siteHasFeature from 'calypso/state/selectors/site-has-feature';
 import { useSiteOption } from 'calypso/state/sites/hooks';
 import { getSelectedSiteId, getSelectedSiteSlug } from 'calypso/state/ui/selectors';
+import { AppState } from 'calypso/types';
 import { SftpCardLoadingPlaceholder } from './sftp-card-loading-placeholder';
 import SshKeys from './ssh-keys';
 
+import './style.scss';
+
 const FILEZILLA_URL = 'https://filezilla-project.org/';
 const SFTP_URL = 'sftp.wp.com';
 const SFTP_PORT = 22;
 
-const SftpClipboardButtonInput = styled( ClipboardButtonInput )( {
-	display: 'block',
-	marginBottom: '16px',
-} );
-
-const SftpQuestionsContainer = styled.div( {
-	marginBottom: '1.5em',
-} );
-
-const SftpPasswordExplainer = styled.p( {
-	marginBottom: '8px',
-} );
-
-const SftpEnableWarning = styled.p( {
-	color: 'var(--color-text-subtle)',
-} );
-
-const SftpSshLabel = styled( FormLabel )( {
-	marginTop: '16px',
-	paddingTop: '16px',
-	borderTop: '1px solid #e0e0e0',
-} );
+type SftpCardComponentProps = SftpCardProps & {
+	translate: LocalizeProps[ 'translate' ];
+	siteId: number | null;
+	siteSlug: string | null;
+	currentUserId: number | null;
+	username: string | null;
+	password: string | null;
+	siteHasSftpFeature: boolean;
+	siteHasSshFeature: boolean;
+	isSshAccessEnabled: boolean;
+	isLoadingSftpData: boolean;
+	createSftpUser: ( siteId: number | null, currentUserId: number | null ) => void;
+	requestSftpUsers: ( siteId: number | null ) => void;
+	resetSftpPassword: ( siteId: number | null, sshUsername: string | null ) => void;
+	requestSshAccess: ( siteId: number | null ) => void;
+	enableSshAccess: ( siteId: number | null ) => void;
+	disableSshAccess: ( siteId: number | null ) => void;
+	removePasswordFromState: ( siteId: number | null, username: string | null ) => void;
+};
```

Comment:
The reason I typically avoid `connect` when converting JS files to TS is to avoid this redundancy. In other words, `useSelector` calls provide us with "free" types, whereas we have to manually respecify types when using `connect`.

I would advise we do the same in this case and employ `useSelector` and `useDispatch` instead of connect. It'll make the code more compact and future refactorings easier.